components/clp-py-utils/clp_py_utils/clp_config.py

```diff
@@ -433,16 +427,29 @@ def validate_staging_directory(cls, field):
             raise ValueError("staging_directory cannot be empty")
         return field
 
+    @root_validator
+    def validate_key_prefix(cls, values):
+        s3_config = values.get("s3_config")
+        if hasattr(s3_config, "key_prefix"):
+            key_prefix = s3_config.key_prefix
+            if "" == key_prefix:
+                raise ValueError("key_prefix cannot be empty")
+            if not key_prefix.endswith("/"):
+                raise ValueError('key_prefix must end with "/"')
+            return values
+        else:
+            raise ValueError("s3_config must have field key_prefix")
```

Comment:
In general, early return is preferred.
```suggestion
        if not hasattr(s3_config, "key_prefix"):
            raise ValueError("s3_config must have field key_prefix")
        key_prefix = s3_config.key_prefix
        if "" == key_prefix:
            raise ValueError("key_prefix cannot be empty")
        if not key_prefix.endswith("/"):
            raise ValueError('key_prefix must end with "/"')
        return values
```